Server/Server.Services/RegistrationService.cs

```diff
@@ -0,0 +1,113 @@
+﻿// <copyright company="Tarcha & Ivchenko Company">
+//      Copyright (c) 2015, All Right Reserved
+// </copyright>
+// <author>Ivan Ivchenko</author>
+// <author>Myroslava Tarcha</author>
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Security.Authentication;
+using System.Security.Principal;
+using System.Threading;
+using Common.DatabaseRepositories;
+using Common.DataContracts;
+using Common.Implementation.Hash;
+
+namespace Server.Services
+{
+    /// <summary>
+    /// Registers new user, saves its account name and password in the DB.
+    /// </summary>
+    public class RegistrationService : IRegistrationService
+    {
+        private const int MinPasswordLength = 6;
+        private readonly IRepository<User> _userRepository;
+        private readonly IHashCalculator _hashCalculator;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RegistrationService"/> class.
+        /// </summary>
+        /// <param name="repositoryFactory">Creates repositories.</param>
+        /// <param name="hashCalculator">The hash calculator.</param>
+        [SuppressMessage("Microsoft.Design", "CA1062:Validate arguments of public methods", MessageId = "0", Justification = "We get it from IoC container.")]
+        public RegistrationService(IRepositoryFactory repositoryFactory, IHashCalculator hashCalculator)
+        {
+            _userRepository = repositoryFactory.CreateRepository<User>();

```

Comment:
Hm... I just thought that it will be nice to refactor Factory so it become a Provider. And also Repository<TEntity, TContext>  in not thread safe, isn't it?